Framework/src/runnerUtils.cxx

```diff
@@ -61,19 +62,35 @@ bool hasChecks(const std::string& configSource)
   return config->getRecursive("qc").count("checks") > 0;
 }
 
+bool is_number(const std::string& s)
+{
+  return !s.empty() && std::find_if(s.begin(),
+                                    s.end(), [](unsigned char c) { return !std::isdigit(c); }) == s.end();
+}
+
 Activity computeActivity(framework::ServiceRegistryRef services, const Activity& fallbackActivity)
 {
-  auto runNumber = computeActivityField<int>(services, "runNumber", fallbackActivity.mId);
-  auto runType = computeActivityField<int>(services, "runType", fallbackActivity.mType);
-  auto run_start_time_ms = computeActivityField<unsigned long>(services, "runStartTimeMs", fallbackActivity.mValidity.getMin());
-  auto run_stop_time_ms = computeActivityField<unsigned long>(services, "runEndTimeMs", fallbackActivity.mValidity.getMax());
+  auto runNumber = computeNumericalActivityField<int>(services, "runNumber", fallbackActivity.mId);
+  std::string runType = services.get<framework::RawDeviceService>().device()->fConfig->GetProperty<std::string>("runType", fallbackActivity.mType);
+  // runType used to be an integer. If we find an integer in a config file, the risk is that it is translated directly to a string (2->"2").
+  // We must rather translate the integer into the corresponding run type string if at all possible.
+  if (is_number(runType)) {
+    try {
+      runType = parameters::GRPECS::RunTypeNames.at(std::stoi(runType));
+    } catch (std::out_of_range& exc) {
+      ILOG(Warning, Ops) << "Activity type was provided as an integer. No matching activity type could be find. Using 'NONE'." << ENDM;
```

Comment:
should we warn also if the text type could be found? I assume we want users to move all the config files to use text run type instead of integer anyway?

Reply:
Yes.